tasks/install.yml

```diff
@@ -35,8 +34,9 @@
   when: not jarfile.stat.exists
 
 - name: 'verify if an update is needed'
-  shell: "ls -1d /opt/artifactory/artifactory-* | grep -v artifactory-{{artifactory_version}}"
-  ignore_errors: yes
+  shell: "ls -1d /opt/artifactory/artifactory-{{artifactory_license}}-* | grep -v artifactory-{{artifactory_license}}-{{artifactory_version}}"
+  failed_when: false
```

Comment:
Combining changed_when and failed_when will cause Ansible to ignore any errors (as ignore_erros did) but it will also cause the task to not log any errors and to be idempotent.